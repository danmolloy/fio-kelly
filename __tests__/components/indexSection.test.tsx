import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import IndexSection, { IndexSectionProps } from "../../components/indexSection";

const mockProps: IndexSectionProps = {
  children: <div></div>,
  title: "Lorem Ipsum",
  classNames: null,
};

describe("<IndexSection />", () => {
  beforeEach(() => {
    render(<IndexSection {...mockProps} />);
  });
  it("index-section is in the document", () => {
    const indexSection = screen.getByTestId("index-section");
    expect(indexSection).toBeInTheDocument();
  });
  it("title is in the document", () => {
    const title = screen.getByText(mockProps.title);
    expect(title).toBeInTheDocument();
  });
  it("matches snapshot", () => {
    const indexSection = screen.getByTestId("index-section");
    expect(indexSection).toMatchSnapshot();
  });
});
