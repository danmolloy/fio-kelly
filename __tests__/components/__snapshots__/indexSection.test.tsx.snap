// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IndexSection /> matches snapshot 1`] = `
<div
  class="flex flex-row w-full"
  data-testid="index-section"
>
  <h1
    class="font-5xl my-8 font-display"
  >
    Lorem Ipsum
  </h1>
</div>
`;
